internal/controller/update_test.go

```diff
@@ -158,8 +160,13 @@ func TestImageUpdateAutomationReconciler_deleteBeforeFinalizer(t *testing.T) {
 		EventRecorder: record.NewFakeRecorder(32),
 	}
 	// NOTE: Only a real API server responds with an error in this scenario.
-	_, err := r.Reconcile(ctx, ctrl.Request{NamespacedName: client.ObjectKeyFromObject(imageUpdate)})
-	g.Expect(err).NotTo(HaveOccurred())
+	g.Eventually(func() bool {
+		_, err := r.Reconcile(ctx, ctrl.Request{NamespacedName: client.ObjectKeyFromObject(imageUpdate)})
+		if err != nil {
+			return false
+		}
+		return true
+	}, timeout).Should(BeTrue())
```

Comment:
```suggestion
	g.Eventually(func() error {
		_, err := r.Reconcile(ctx, ctrl.Request{NamespacedName: client.ObjectKeyFromObject(imageUpdate)})
		return err
	}, timeout).Should(Succeed())
```

This way the failure message actually tells us what failed.